src/adapter/src/catalog.rs

```diff
@@ -1641,9 +1650,14 @@ struct AllocatedBuiltinSystemIds<T> {
     new_builtins: Vec<(T, GlobalId)>,
     migrated_builtins: Vec<(GlobalId, u64)>,
 }
-
+/// Objects are uniquely defined by the (schema_name, object_type, object_name)
+/// tuple.
+/// As such, a name can only be used once for a given schema and object type.
+/// The same name can be used for multiple catalog item types, such as for a
+/// type and a function.
```

Comment:
Wait, this explanation doesn't seem quite right to me? You can't have a table and an index with the same name, for example, even though they're different types, right?

Reply:
Yes, that is what the check we added to `insert_item` does.